src/Npgsql/NpgsqlConnectionStringBuilder.cs

```diff
@@ -683,6 +683,24 @@ public bool IncludeErrorDetail
     }
     bool _includeErrorDetail;
 
+    /// <summary>
+    /// When enabled, failed statements are included on <see cref="NpgsqlException.BatchCommand" />. These can contain sensitive data.
+    /// </summary>
+    [Category("Security")]
+    [Description("When enabled, failed statements are included on NpgsqlException.BatchCommand. These can contain sensitive data.")]
+    [DisplayName("Include Failed Statement")]
+    [NpgsqlConnectionStringProperty]
+    public bool IncludeFailedStatement
```

Comment:
nit: given that the property is `NpgsqlException.BatchCommand`, maybe call this IncludeFailedBatchedCommand?

```suggestion
    public bool IncludeFailedBatchedCommand
```